diffrax/nonlinear_solver/newton.py

```diff
@@ -82,7 +82,8 @@ class NewtonNonlinearSolver(AbstractNonlinearSolver):
 
     def __post_init__(self):
         if self.max_steps is not None:
-            error_if(self.max_steps < 2, "max_steps must be at least 2.")
+            if self.max_steps < 2:
```

Comment:
These two `if` statements can be combined.

Reply:
Thanks Patrick. We were thinking that you wanted to keep `max_steps` being a `Tracer` value and that makes sense. I can verify though that this PR solves tests breaking for our usecase.